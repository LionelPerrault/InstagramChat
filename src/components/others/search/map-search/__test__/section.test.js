import React from 'react'
import { create } from 'react-test-renderer'
import SearchSection from '../section'

describe('Search Section Component', () => {
  const map = list => list.map(l => <span key={l}>{l}</span>)

  it('should match snapshot for users search', () => {
    const tree = create(
      <SearchSection searchList={map(['first', 'second'])} listFor="member" />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should match snapshot for hashtags search', () => {
    const tree = create(
      <SearchSection
        searchList={map(['#first', '#second'])}
        listFor="hashtag"
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should match snapshot for group search', () => {
    const tree = create(
      <SearchSection
        searchList={map(['normal-grp', 'crazy-group'])}
        listFor="group"
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
