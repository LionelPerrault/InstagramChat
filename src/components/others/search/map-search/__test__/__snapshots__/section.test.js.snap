// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Section Component should match snapshot for group search 1`] = `
<div
  className="s_d"
>
  <span
    className="s_header"
  >
    2 groups
  </span>
  <span>
    normal-grp
  </span>
  <span>
    crazy-group
  </span>
</div>
`;

exports[`Search Section Component should match snapshot for hashtags search 1`] = `
<div
  className="s_d"
>
  <span
    className="s_header"
  >
    2 hashtags
  </span>
  <span>
    #first
  </span>
  <span>
    #second
  </span>
</div>
`;

exports[`Search Section Component should match snapshot for users search 1`] = `
<div
  className="s_d"
>
  <span
    className="s_header"
  >
    2 members
  </span>
  <span>
    first
  </span>
  <span>
    second
  </span>
</div>
`;
